### PR TITLE
KAFKA-7678: Avoid NPE when closing the RecordCollector

### PR DESCRIPTION
**Pull request by jonathansantilli**
Check if the Producer is not null to avoid calling **producer.close()** on a **null** value.

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java]: Nit: use `producer != null`

**Review by mjsax — COMMENTED** [review]
Overall LGTM. One nit.

Call for second review anyof @guozhangwang @bbejeck @vvcephei

**Review by bbejeck — COMMENTED** [review]
LGTM as well modulo the comment from @mjsax

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java
Comment (jonathansantilli) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java]: Done

**Comment by jonathansantilli** [comment]
Thanks for the review @mjsax @bbejeck, code updated as suggested :)
Cheers!

**Review by bbejeck — APPROVED** [review]
@jonathansantilli thanks for the update and the contribution!

**Comment by jonathansantilli** [comment]
It was a pleasure @bbejeck, hope this goes to the master branch soon to test it.
Cheers!

**Comment by guozhangwang** [comment]
@jonathansantilli it's already merged to trunk by @mjsax , you can check out latest trunk head and try it out now :) Thanks for your contributions again!

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `2.1`, `2.0`, and `1.1` (not required for older versions).

**Comment by mjsax** [comment]
Thanks for the patch @jonathansantilli

**Comment by jonathansantilli** [comment]
It was a pleasure @guozhangwang @mjsax

**Pull request merged**
